pkg/engine/mutate/patchesUtils.go

```diff
@@ -34,6 +33,123 @@ func generatePatches(src, dst []byte) ([][]byte, error) {
 	return patchesBytes, err
 }
 
+// sortRemovalPatches sort the removal patches(with same path) by the key of index in descending order
+//
+// [{"op":"remove","path":"/a/b/0"},{"op":"remove","path":"/a/b/1"}]
+// will be reordered to
+// [{"op":"remove","path":"/a/b/1"},{"op":"remove","path":"/a/b/0"}]
+func sortRemovalPatches(originalPatches []jsonpatch.JsonPatchOperation) []jsonpatch.JsonPatchOperation {
```

Comment:
seems like this method does more than sort removal patches - it also removes patches, right?

Can we please rename to clarify the intent and update the comment?